### ISSUE TITLE
GIKT: A Graph-based Interaction Model for Knowledge Tracing

### ISSUE THREAD

**Issue by hello-im-yj** [One-Day-One-Paper/One-Day-One-Paper#125]
***Abstract***

purpose : From the model perspective, previous models can hardly capture the **long-term dependency** of student exercise history, and cannot model the interactions between student-questions, and student-skills in a consistent way.

**GIKT Model**

- utilizes graph convolutional network (GCN) to substantially incorporate question-skill correlations via embedding propagation.
    <img width="455" alt="스크린샷 2022-03-27 오후 11 51 26" src="https://user-images.githubusercontent.com/63237947/160287174-4c52655b-5c13-40a0-86c3-33a320b281e7.png">
    
- { the student’s current state, the student’s history states, the target question and related skills }으로 학생의 질문 숙련 정도를 일반화함.
- Experiments : SOTA Performance with 2%–6% absolute AUC improvement.

**4 The Proposed Method GIKT**

![image](https://user-images.githubusercontent.com/63237947/160287366-d09f8485-428a-4507-a305-0e275bca7758.png)

{student’s current state, the selected student’s history states, the target question and related skills} → for the prediction pt.

- recap module → $h_{2},h_{t-1}$...  관련된 history state를 가져오는 역할
    - hard : s1,s2와 관련된 question으로부터 나온 ht 모두 가져옴
    - soft : attention score가 높은 top-K개를 가져오도록 학습시킴
- RNN으로 ht를 모델링한 결과 ht
- relate된 question 과 skills

4.5 Generalized Interaction Module 

![image](https://user-images.githubusercontent.com/63237947/160287606-0a5fcb9d-f9e3-4a46-b8ac-2f913b43bb8b.png)

$f_i$ = 관련된 $h_{(1\ to\  t-1)} + h_t$
$f_j$ = $q_t$ 의 이웃 skill embedding + $q_t$
-> attention score를 training
$g$ : inner product → 결과 : 스칼라 

& optimization : update the parameters in our model using gradient descent by minimizing the cross entropy loss.
